packages/template-retail-react-app/app/components/login/index.jsx

```diff
@@ -8,21 +8,15 @@
 import React, {Fragment} from 'react'
 import PropTypes from 'prop-types'
 import {FormattedMessage} from 'react-intl'
-import {
-    Alert,
-    Box,
-    Button,
-    Stack,
-    Text
-} from '@salesforce/retail-react-app/app/components/shared/ui'
+import {getConfig} from '@salesforce/pwa-kit-runtime/utils/ssr-config'
```

Comment:
I'm guessing that this component in multiple places? 

🤔  Typically you'd want to make something like this a dependency that you can inject (ask chatGPT why dependency is a good thing). 

You'll be able to test this component without having to mock that module and the component will become more portable. 

Imagine a world where we want to create a component library, this would mean that this component could only be used in a pwa-kit project. Not the worst thing in the world, but not the best. 

Feel free to push back and we can discuss.

Reply:
That's a good point. We tried to put the `getConfig` at the highest login component so it would only have to be called once. 

I see that this `LoginForm` component is rendered in the:
- `Login` component and the
- `AuthModal` component which is rendered in `App`, `ContactInfo`, and `withRegistration` 

I guess our thought process was if we put it higher up we'd need to call `getConfig` from many different places and continue to pass the value down to the login UI components.

Reply:
I see that in other components, a hook is used to access `getConfig` instead of calling it directly in the component. For example in the `useEinstein` hook.

Would it be better if we add something like a `useAdditionalLogin` hook that accesses getConfig and returns the values that way we don't have to mock the entire config?

Reply:
Just because we did it somewhere else doesn't make it right haha... but in general I want you to think about how your'd develop components to be reuseable. A good exercise is is to say, will this component work outside of a pwa-kit project if someone wanted to use it" .. the answer here would be no because it doesn't know what this configuration util is, so it would fail at that point

Reply:
Gotcha thanks for explaining! I refactored the code so that the `LoginForm` component gets the configuration from the props. And instead `Login` and `useAuthModal` are where `import {getConfig} from '@salesforce/pwa-kit-runtime/utils/ssr-config'` is used instead